contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,807 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import "./System.sol";
+import "./lib/Utils.sol";
+import "./interface/IBSCValidatorSet.sol";
+import "./interface/IGovToken.sol";
+import "./interface/IStakeCredit.sol";
+
+contract StakeHub is System, Initializable {
+    using Utils for string;
+    using Utils for bytes;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /*----------------- constant -----------------*/
+    uint256 private constant BLS_PUBKEY_LENGTH = 48;
+    uint256 private constant BLS_SIG_LENGTH = 96;
+
+    address private constant DEAD_ADDRESS = address(0xdead);
+
+    //TODO
+    bytes private constant INIT_BC_CONSENSUS_ADDRESSES =
+        hex"00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000";
+    bytes private constant INIT_BC_VOTE_ADDRESSES =
+        hex"00000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000";
+
+    /*----------------- storage -----------------*/
+    bool private _stakingPaused;
+
+    uint256 public transferGasLimit;
+
+    // stake params
+    uint256 public minSelfDelegationBNB;
+    uint256 public minDelegationBNBChange;
+    uint256 public maxElectedValidators;
+    uint256 public unbondPeriod;
+
+    // slash params
+    uint256 public downtimeSlashAmount;
+    uint256 public doubleSignSlashAmount;
```

Comment:
`doubleSignSlashAmount` and `doubleSignJailTime` are used in both malicious vote and double sign, so I think it is better to have another name: `felonyJailTime`, `felonySlashAmount`

Reply:
Done